libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -179,6 +186,51 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+// Preliminary implementation that doesn't take care of many things like
+// name collisions etc. Mainly used for test cases.
+bool IRGeneratorForStatements::visit(InlineAssembly const& _inlineAsm)
+{
+	struct PrefixNameDispenser: public yul::NameDispenser
```

Comment:
I'm wondering whether it would be shorter and simpler to implement a new class that is similar to the BodyCopier but which only implements `translateIdentifier` where it looks up the identifier in the external references - if it is a match it does what is done in line 212. If there is no match, it returns `"usr$" + name`.